aspnetcore/security/authorization/roles.md

```diff
@@ -3,92 +3,141 @@ title: Role-based authorization in ASP.NET Core
 author: rick-anderson
 description: Learn how to restrict ASP.NET Core controller and action access by passing roles to the Authorize attribute.
 ms.author: riande
+monikerRange: '>= aspnetcore-3.1'
 ms.date: 10/14/2016
 no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
 uid: security/authorization/roles
 ---
 # Role-based authorization in ASP.NET Core
 
+::: moniker range=">= aspnetcore-6.0"
+
+<a name="security-authorization-role-based"></a>
+
+When an identity is created it may belong to one or more roles. For example, Tracy may belong to the `Administrator` and `User` roles while Scott may only belong to the `User` role. How these roles are created and managed depends on the backing store of the authorization process. Roles are exposed to the developer through the [IsInRole](/dotnet/api/system.security.principal.genericprincipal.isinrole) method on the [ClaimsPrincipal](/dotnet/api/system.security.claims.claimsprincipal) class. [AddRoles](/dotnet/api/microsoft.aspnetcore.identity.identitybuilder.addroles#Microsoft_AspNetCore_Identity_IdentityBuilder_AddRoles__1) must be added to Role services.
+
+We recommend not using Roles as claims, but rather using a [claims](xref:security/authorization/claims). The Single Page Apps (SPAs) templates for ASP.NET Core are not designed to use Roles. To secure SPAs, see <xref:security/authentication/identity/spa>.
```

Comment:
@blowdart  please review this paragraph.

Reply:
"We recommend not using Roles as claims, but rather using a [claims](xref:security/authorization/claims)." I can't figure out what you're trying to say here. 

I'm going to guess we're saying

While Roles are claims, not all claims are roles. Depending on the identity issuer a role may be a collection of users that may apply claims for group members, as well as an actual claim on an identity. However, claims are meant to be information about an individual user and using roles to add claims to a user can confuse the boundary between the user and their individual claims. 

Maybe?

Reply:
@blowdart see [this comment](https://github.com/dotnet/AspNetCore.Docs/issues/14944#issuecomment-619263127)

@blowdart can elaborate, but the basic idea is that we are recommending/encouraging moving away from Roles as claims towards a full claim based approach, so that's why the new SPA templates do not make it easy to continue using the old style roles approach. @Rick-Anderson do you think we can add a snippet to this effect somewhere appropriate in the docs?

Reply:
Ah!

OK, 

While roles are claims, not all claims are roles. Depending on the identity issuer a role may be a collection of users that may apply claims for group members, as well as an actual claim on an identity. However, claims are meant to be information about an individual user and using roles to add claims to a user can confuse the boundary between the user and their individual claims. This confusion is why the SPA templates are not designed around roles. In addition, for organizations migrating from an on-premises legacy system the proliferation of roles over the years can mean a role claim may be too large to be contained within a token usable by SPAs.

Reply:
@blowdart  added that. Split up one sentence into two. Ready for review.